src/scala/scripts/ScalaPBGenerator.scala

```diff
@@ -0,0 +1,68 @@
+package scripts
+
+import io.bazel.rulesscala.jar.JarCreator
+import io.bazel.rulesscala.io_utils.DeleteRecursively
+import java.io.{ File, PrintStream }
+import java.nio.file.{ Files, Path, Paths }
+import scala.collection.mutable.Buffer
+import io.bazel.rulesscala.worker.{ GenericWorker, Processor }
+import scala.io.Source
+import protocbridge.ProtocBridge
+import scalapb.ScalaPbCodeGenerator
+import com.trueaccord.scalapb.{Config, ScalaPBC, ScalaPbcException}
+
+object ScalaPBWorker extends GenericWorker(new ScalaPBGenerator) {
+
+  override protected def setupOutput(ps: PrintStream): Unit = {
+    System.setOut(ps)
+    System.setErr(ps)
+    Console.setErr(ps)
+    Console.setOut(ps)
+  }
+
+  def main(args: Array[String]) {
+    try run(args)
+    catch {
+      case x: Exception =>
+        x.printStackTrace()
+        System.exit(1)
+    }
+  }
+}
+
+class ScalaPBGenerator extends Processor {
+  def deleteDir(path: Path): Unit =
+    try DeleteRecursively.run(path)
+    catch {
+      case e: Exception => ()
+    }
+
+  def processRequest(args: java.util.List[String]) {
+    val jarOutput = args.get(0)
+    val protoFiles = args.get(1).split(':').toList
+
+    val tmp = Paths.get(Option(System.getenv("TMPDIR")).getOrElse("/tmp"))
```

Comment:
I think we actually want:

`System.getProperty("java.io.tmpdir")`

Which I think bazel sets up correctly.

we got burned on this somewhere else in some context. Did you copy this from another rule?

Reply:
Yeah I copied this from the scrooge rules. Want me to fix that one too?